components/brave_rewards/browser/file_util.cc

```diff
@@ -0,0 +1,151 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_rewards/browser/file_util.h"
+
+#include <memory>
+
+#include "base/logging.h"
+#include "base/strings/string_util.h"
+
+namespace brave_rewards {
+
+namespace {
+
+bool SeekLine(
+    base::File* file,
+    int64_t* offset) {
+  DCHECK(file);
+  DCHECK(offset);
+
+  char buffer[1];
+
+  do {
+    if (file->ReadAtCurrentPos(buffer, 1) == -1) {
+      return false;
+    }
+
+    const int64_t seek_position = file->Seek(base::File::FROM_CURRENT, -2);
+    if (seek_position == -1) {
+      return false;
+    }
+
+    *offset = seek_position;
+  } while (buffer[0] != '\n');
+
+  return true;
+}
+
+int64_t SeekNumLines(
+    base::File* file,
+    const uint64_t num_lines) {
+  DCHECK(file);
+
+  if (num_lines == 0) {
+    return 0;
+  }
+
+  const int64_t length = file->GetLength();
+  if (length == -1) {
+    return -1;
+  }
+
+  if (length == 0) {
+    return 0;
+  }
+
+  if (file->Seek(base::File::FROM_END, -1) == -1) {
+    return -1;
+  }
+
+  uint64_t line_count = 0;
+
+  int64_t offset = length - 1;
+
+  do {
+    if (!SeekLine(file, &offset)) {
+      return 0;
+    }
+
+    line_count++;
+  } while (line_count < num_lines + 1);
+
+  return offset + 2;  // Position the offset after the line feed
+}
+
+bool TruncateFile(
+    base::File* file,
+    const int64_t offset) {
+  DCHECK(file);
+
+  if (offset == 0) {
```

Comment:
Seems like we could just wrap `base::TruncateFile` instead?

Reply:
We are truncating the file from the end, not from the start

Reply:
I see, maybe it should be named `TruncateFileFromEnd` or something like that since it's atypical.

Reply:
Fixed